src/utilities/controllers/discon_rotor_test_controller.cpp

```diff
@@ -15,28 +15,34 @@ void PITCH_CONTROLLER(
     char* const avcMSG
 ) {
     // Map swap from calling program to struct
-    ControllerIO* io = reinterpret_cast<ControllerIO*>(avrSWAP);
+    ControllerIO io;
+    auto swap_array = std::array<float, kSwapArraySize>{};
+    std::copy(avrSWAP, std::next(avrSWAP, 81), swap_array.begin());
+    io.CopyFromSwapArray(swap_array);
 
     // Update pitch angle in radians
-    io->pitch_command_1 = 2. * M_PI * io->time / 3.;  // Full rotation every 3 seconds
-    io->pitch_command_2 = 2. * M_PI * io->time / 6.;  // Full rotation every 6 seconds
-    io->pitch_command_3 = 2. * M_PI * io->time / 9.;  // Full rotation every 9 seconds
+    io.pitch_command_1 = 2. * M_PI * io.time / 3.;  // Full rotation every 3 seconds
+    io.pitch_command_2 = 2. * M_PI * io.time / 6.;  // Full rotation every 6 seconds
+    io.pitch_command_3 = 2. * M_PI * io.time / 9.;  // Full rotation every 9 seconds
 
     // Set failure flag to zero
     *aviFAIL = 0;
 
     // Set message to success
-    strncpy(avcMSG, "success\0", io->message_array_size);
-    avcMSG[static_cast<int>(io->message_array_size) - 1] = 0;
+    strncpy(avcMSG, "success\0", io.message_array_size);
```

Comment:
`strncpy` alone does not guarantee that the string will be null terminated; hence the following line. https://stackoverflow.com/questions/1453876/why-does-strncpy-not-null-terminate

Reply:
Doesn't the `\0` take care of this already?

Reply:
If 'avcMSG' is shorter than 'success' it will fill the buffer with characters but not add the null termination. This is one of the issues with C string functions.

Reply:
Fair enough.  I've added a version of the fix back in that clang-tidy seems to like.  

If you find this version objectionable, I might just turn off the check which complains about pointer arithmetic and return to the old fix.  These things are supposed to make it easier for us to write good code, not make us jump through hoops.